websocket/core/src/main/java/io/undertow/websockets/WebsocketConnectionBuilder.java

```diff
@@ -214,6 +217,15 @@ public void operationComplete(ChannelFuture future) throws Exception {
         return handler.handshakeFuture;
     }
 
+    // the bootstrap always uses NioSocketChannel, so we need to provide a proper NioEventLoopGroup
+    private EventLoopGroup determineEventLoopGroup() {
+        if (NioEventLoopGroup.class.equals(eventLoopGroup.getClass())) {
+            return eventLoopGroup;
+        }
+        // there is not much else we can do here...
+        return new NioEventLoopGroup();
```

Comment:
I agree, but it's unfortunate. We should add a note in the Quarkus documentation when using the legacy web socket implementation; native transport is not recommended.